### ISSUE TITLE
compress job stdout files (job_out.*.*.txt) in webdir

### ISSUE THREAD

**Issue by belforte** [dmwm/CRABServer#5517]
job stdout linger along until we declare the task end of life and remove the web directory, tipycally 38 days after intial submission. See

Only a small fraction of those files is ever used (to check on issues) and yet they use the major part of disk space on /home/grid. Tipycal file size is from 50 to 300KB (now that we do not print tarball content anymore). But large tasks with several resubmissions can have O(50k) files, i.e. 15GB per task. It does not take much to have few tens of such tasks in a month period bringing us close to the 1TB mark.

So we entertain the possibiliyt of zipping those stdout, which shrinks them a factor 5~10.

This is technically straightforward. Just need to replace this line
```
shutil.copy(stdout_tmp, fname)
```
in   https://github.com/dmwm/CRABServer/blob/55cf1a64b69f8211c0d337000f80fb98ae096c8c/src/python/TaskWorker/Actions/PostJob.py#L1380

with what python doc suggests in https://docs.python.org/2/library/gzip.html 
```
import gzip
with open(stdout_tmp, 'rb') as f_in, gzip.open(fname+'.gz', 'wb') as f_out:
    shutil.copyfileobj(f_in, f_out)
```

I.e. the stdout comes uncompressed via HTCondor to the spool directory, then the PostJob reames it to add the retry number and moves it to the webdir, it could compress it at that time.

The problem is whether we really want to do it since it is a non traspsarent operation for the user.
I did a few test with opening a gzipped text file in firefox, but it seems the only way it works is by putting the .gz extension in the file name. 

So we will need to:
1 change the URL's to job log reported to dashboard so that they point to a file with .gz extension
1 be prepared for having to go through the archive manager when opening a stdout file in the browser, and may still need the Open in Browser firefox extension. Even if the web says that firefox does it automatically when the server is properly configured, our server does not look so.
1 modify crab getlog to deal with zipped files, maybe have the option to unzip automatically when copying to local UI, or just do this by default so it stays transparent

IN other words, I'd be prepared to deal with gzipped stdout, but am not sure about others and do not want to spend time teaching all users.

In light of those, I propose to sit on this until it is clear that we need to. All in all we can also gain a factor 2 by expanding the /home/grid partitions, but going for O(20) schedd's with O(10) TB each for files none looks at, seems too much.

Maybe when it gets really needed, we can get cmsweb to properly present `*gz` files with the header
```
Content-Type: text/html
Content-Encoding: gzip
```
(at least IIUC).

**Comment by belforte** [comment]
I anybody has ideas on how to solve the drawbacks listed above (esp. 2.) so that we can instead to ahead an do this "now", please add your wisdom here
